File: C/src/Sully.c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>
#include <stdlib.h>
int main(){
int i = 5;
if (i >= 0){
int f;
char path[20];
sprintf(path, "Sully_%d.c", i);
i--;
f = open(path,O_CREAT|O_WRONLY|O_TRUNC, S_IRWXU);
char src[500] = "#include <stdio.h>%1$c#include <fcntl.h>%1$c#include <unistd.h>%1$c#include <stdlib.h>%1$cint main(){%1$cint i = %2$d;%1$cif (i >= 0){%1$cint f;%1$cchar path[20];%1$csprintf(path, %3$cSully_%%d.c%3$c, i);%1$ci--;%1$cf = open(path,O_CREAT|O_WRONLY|O_TRUNC, S_IRWXU);%1$cchar src[5000] = %3$c%4$s%3$c;%1$cdprintf(f,src,10,i,34,src);%1$cchar cmd[100];%1$csprintf(cmd, %3$cgcc -Wall -Werror -Werror -o Sully_%%1$d Sully_%%1$d.c && ./Sully_%%1$d%3$c, i+1);%1$csystem(cmd);%1$cclose(f);%1$c}%1$c}%1$c";
dprintf(f,src,10,i,34,src);
char cmd[100];
sprintf(cmd, "gcc -Wall -Werror -Werror -o Sully_%1$d Sully_%1$d.c && ./Sully_%1$d", i+1);
system(cmd);
close(f);
}
}

File: PYTHON/Sully.py
import os
i = 5

if (i >= 0):
    f = open("Sully_"+str(i)+".py", "w+")
    i -= 1
    s1 = "import os{0}i = {2}{0}if (i >= 0):{0}    "
    s2 = "f = open({1}Sully_{1}+str(i)+{1}.py{1}, {1}w+{1}){0}    i -= 1{0}    "
    s3 = "s1 = {1}{3}{1}{0}    s2 = {1}{4}{1}{0}    s3 = {1}{5}{1}{0}    s4 = {1}{6}{1}{0}    s5 = {1}{7}{1}{0}    "
    s4 = "f.write((s1+s2+s3+s4+s5).format(chr(10), chr(34), i, s1, s2, s3, s4, s5)){0}    "
    s5 = "f.close(){0}    os.system({1}python Sully_{1}+str(i+1)+{1}.py{1}){0}"
    f.write((s1+s2+s3+s4+s5).format(chr(10), chr(34), i, s1, s2, s3, s4, s5))
    f.close()
    os.system("python Sully_"+str(i+1)+".py")
